recipe/activate-mdt.sh

```diff
@@ -1,20 +1,54 @@
-if [ -z "$MACOSX_DEPLOYMENT_TARGET" ]; then
-    export MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
-else
+#!/usr/bin/env bash
+
+if [ -n "$MACOSX_DEPLOYMENT_TARGET" ]; then
     export CONDA_SYSROOT_@PLATFORM@_BACKUP_MACOSX_DEPLOYMENT_TARGET="${MACOSX_DEPLOYMENT_TARGET}"
-    export MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
 fi
+export MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
 
-if [ -z "${CMAKE_ARGS}" ]; then
-    export CMAKE_ARGS="-DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
+# Priority order for SDK setup:
+# 1. If OSX_SDK_DIR is set, use it to set both CONDA_BUILD_SYSROOT and SDKROOT
+# 2. Else if CONDA_BUILD_SYSROOT is set, use it and set SDKROOT to match
+# 3. Else if SDKROOT is set, use it and set CONDA_BUILD_SYSROOT to match
+# 4. Else warn that none are set
+
+if [ -n "${OSX_SDK_DIR}" ]; then
+    # OSX_SDK_DIR is set, back up the others if we have them and set them both to the new path.
+    if [ -n "${CONDA_BUILD_SYSROOT}" ]; then
+        export CONDA_SYSROOT_@PLATFORM@_BACKUP_CONDA_BUILD_SYSROOT="${CONDA_BUILD_SYSROOT}"
+    fi
+    if [ -n "${SDKROOT}" ]; then
+        export CONDA_SYSROOT_@PLATFORM@_BACKUP_SDKROOT="${SDKROOT}"
+    fi
+    export CONDA_BUILD_SYSROOT="${OSX_SDK_DIR}/MacOSX${MACOSX_DEPLOYMENT_TARGET}.sdk"
+    export SDKROOT="${CONDA_BUILD_SYSROOT}"
+elif [ -n "${CONDA_BUILD_SYSROOT}" ]; then
+    # We have a SYSROOT and no SDKROOT, use the SYSROOT to set the SDKROOT.
+    export SDKROOT="${CONDA_BUILD_SYSROOT}"
```

Comment:
```suggestion
    # Use existing CONDA_BUILD_SYSROOT
    if [ -n "${SDKROOT}" ]; then
        export CONDA_SYSROOT_@PLATFORM@_BACKUP_SDKROOT="${SDKROOT}"
    fi
    export SDKROOT="${CONDA_BUILD_SYSROOT}"
```
SDKROOT may have been set beforehand.